src/Microsoft.AspNet.Mvc.TagHelpers/Internal/PartialAttributeLoggerStructureOfT.cs

```diff
@@ -11,49 +11,48 @@
 namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
 {
     /// <summary>
-    /// An <see cref="ILogValues"/> for log messages regarding <see cref="ITagHelper"/> instances that opt out of
+    /// Log values for <see cref="ITagHelper"/> instances that opt out of
     /// processing due to missing attributes for one of several possible modes.
     /// </summary>
-    public class PartialModeMatchLoggerStructure<TMode> : PartialModeMatchLoggerStructure
+    public class PartialModeMatchLogValues<TMode> : ILogValues
     {
         private readonly string _uniqueId;
         private readonly string _viewPath;
         private readonly IEnumerable<ModeMatchAttributes<TMode>> _partialMatches;
 
         /// <summary>
-        /// Creates a new <see cref="PartialModeMatchLoggerStructure{TMode}"/>.
+        /// Creates a new <see cref="PartialModeMatchLogValues{TMode}"/>.
         /// </summary>
         /// <param name="uniqueId">The unique ID of the HTML element this message applies to.</param>
         /// <param name="viewPath">The path to the view.</param>
         /// <param name="partialMatches">The set of modes with partial required attributes.</param>
-        public PartialModeMatchLoggerStructure(
+        public PartialModeMatchLogValues(
             string uniqueId,
             string viewPath,
             [NotNull] IEnumerable<ModeMatchAttributes<TMode>> partialMatches)
-            : base(values: new Dictionary<string, object>
-            {
-                ["UniqueId"] = uniqueId,
-                ["ViewPath"] = viewPath,
-                ["PartialMatches"] = partialMatches
-            })
         {
             _uniqueId = uniqueId;
             _viewPath = viewPath;
             _partialMatches = partialMatches;
         }
-        
-        /// <summary>
-        /// Generates a human readable string for this structured log message.
-        /// </summary>
-        /// <returns>The message.</returns>
-        public override string Format()
+
+        public override string ToString()
         {
             var newLine = Environment.NewLine;
             return string.Format(
-                $"Tag Helper with ID {_uniqueId} in view '{_viewPath}' had partial matches while determining mode:{newLine}\t{{0}}",
+                $"Tag Helper with ID '{_uniqueId}' in view '{_viewPath}' had partial matches " +
+                $"while determining mode:{newLine}\t{{0}}",
                     string.Join($"{newLine}\t", _partialMatches.Select(partial =>
                         string.Format($"Mode '{partial.Mode}' missing attributes:{newLine}\t\t{{0}} ",
                             string.Join($"{newLine}\t\t", partial.MissingAttributes)))));
         }
+
+        public IEnumerable<KeyValuePair<string, object>> GetValues()
+        {
+            yield return new KeyValuePair<string, object>("Message", "Tag Helper has missing required attributes.");
```

Comment:
align this `Message` better with the `ToString()` value e.g. "Tag helper had partial matches while determining mode."